extractor/extractor.py

```diff
@@ -0,0 +1,195 @@
+
+    def clean_soup(self):
+        decomposable_tags = self.soup.find_all(['script', 'noscript', 'style', 'header', 'footer', 'meta', 'link'])
+        comments = self.soup.find_all(text=lambda text: isinstance(text, Comment))
+        ads = self.soup.find_all(re.compile('.*'), {'class': pattern_for_ads})
+
+        comments = [comment.extract() for comment in comments]
+        [tag.decompose() for tag in decomposable_tags]
+        [ad.decompose() for ad in ads]
+
+    def extract_data_from_html(self):
+        # Extract data from head
+        head = self.soup.find('head')
+        self.extract_data_from_html_head(head)
+
+        # Extract data from body
+        body_soup = self.soup.find('body')
+        self.extract_data_from_html_body(body_soup)
+
+    def extract_data_from_html_head(self, soup_head):
+        title = soup_head.title
+        text_type = 'title'
+        text_string = title.get_text()
+        _content = contents.Text(text_string, text_type)
+        self.contents_list.add_content(_content)
+
+    def extract_data_from_html_body(self, node):
+        isContent, _content = self.is_content(node)
+
+        # Add to the final list if a valid content is extracted
+        if isContent:
+            self.contents_list.add_content(_content)
+
+        if node is not None and node.name is not None and not isContent:
+            for child in node.contents:
+                self.extract_data_from_html_body(child)
+
+    def is_content(self, node):
+        # checks if content is text
+        isText, text = self.is_text(node)
+        if isText:
+            return True, text
+
+        # checks if content is image
+        isImage, image = self.is_image(node)
+        if isImage:
+            return True, image
+
+        # checks if content is video
+        isVideo, video = self.is_video(node)
+        if isVideo:
+            return True, video
+
+        # TODO: Move extraction for each content type to individual module
+        # TODO: Add extractors for other content types too
+
+        return False, {}
+
+    def empty_text(self, _str):
+        _str = _str.strip('\n')
+        return _str == '\n' or _str == ' ' or len(_str) == 0
+
+    def not_important_text(self, node):
+        if 0 < len(node.get_text()) < 50 and node.name not in heading_tags:
+            return True
+        else:
+            return False
+
+    def is_text(self, node):
+        if type(node) == bs4.element.Tag:
+            text_data = node.get_text().strip()
+            if (node.name in all_text_tags) and not self.empty_text(text_data) and not self.not_important_text(node):
+                text_type = node.name
+                text = contents.Text(text_data, text_type)
+                return True, text
+        elif type(node) == bs4.element.NavigableString and not self.empty_text(node) and len(node) > 50:
+            node = node.strip()
+            text_data = node
+            text = contents.Text(text_data, '')
```

Comment:
We have constructor with one single argument also.

Reply:
Will be updated in next iteration. Thanks for the suggestion!